File: code/java/Basics.java
import java.util.Scanner;
import java.io.PrintStream;
import java.util.HashMap;
import java.util.Map;
import java.util.List;
import java.util.ArrayList;
public class Basics{
    public static void main(String[] args){
        // System.out.println("hello world");
        

        //data types
        // char ch='a';
        // byte bite=3;
        // short short_integer=33;
        // int integer=32;
        // long long_int=4l;
        // float floating_point=4.332f;
        // double double_precision=34.44444;
        // boolean bool=true;

        
        
        // typecasting and taking inputs
        // Scanner scan=new Scanner(System.in);
        // char ch=scan.next().charAt(0);
        // byte bite=scan.nextByte();
        // short short_integer=scan.nextShort();
        // int integer=scan.nextInt();
        // long long_int=scan.nextLong();
        // float floating_point=scan.nextFloat();
        // double double_precision=scan.nextDouble();
        // boolean bool=scan.nextBoolean();
        // String str=scan.nextLine();
        // System.out.println(ch+" "+bite+" "+ short_integer+" "+integer+" "+long_int+" "+floating_point+" "+double_precision+" "+bool);
        // System.out.println(str);
        // PrintStream p=System.out;
        // p.println((int)3.14);
        // p.println((char)93);
        // int n=Integer.parseInt("44");
        // p.println(n*n);
        // String s=Integer.toString(n*n);
        // p.println(s);
        // for(int i=-3;i<10;i++){
        //     p.println(factorial(i));
        // }
        
        // Map<String,Integer> map=new HashMap<>();
        // map.put("name0",3);
        // map.put("name1",4);
        // map.put("name2",5);
        // p.println(map.get("name7"));//returns null
        // p.println(map.get("name0"));
        // p.println(map.getOrDefault("name7",-1));
        // p.println(map.keySet());
        // p.println(map.values());
        // p.println(map.entrySet());
        // p.println(map.keySet().contains("name99"));
        // map.remove("name0");
        // p.println(map.keySet());


        // List<String> list=new ArrayList<>();
        // list.add("hi");
        // list.add("hiff");
        // list.add("higgd");
        // list.add("hidfd");
        // list.add("hiii");
        // list.add(3,"hello");
        // list.remove("hiff");
        // list.remove(0);

        // p.println(list);
        // p.println(list.size());
        // list.get(list.size()-1);
        // p.println(list.contains("hiff"));




        




        


        
    }
    public static long factorial(int positive_integer){
        if (positive_integer<0){
            return -1;
        }
        long result=1;
        for(int i=1;i<=positive_integer;++i){
            result*=i;
        }
        return result;
    }
}

File: code/c++/basics.cpp
#include<iostream>
#include<string>
#include<vector>
#include<map>
using namespace std;
long factorial(int);
int main(){
    // cout<<"hello world!"<<endl;
    // short short_integer;
    // short_integer=33;
    // cout<<short_integer;
    // int integer=33;
    // long long_integer=34;
    // unsigned short us=33;
    // unsigned int ui=234;
    // unsigned long ul=44;
    // cout<<integer<<long_integer<<us<<ui<<ul<<endl;
    // char ch='a';
    // cout<<ch;//prints the character
    // cout<<ch+3;//prints the integer
    // float floating_point=3.3f;
    // double double_precision=4.444;
    // cout<<floating_point;
    // cout<<double_precision;

    // bool is_there=false;
    // cout<<is_there;//prints 1 and 0 for booleans

    // string str="hi there beautiful";
    // string st;
    // st="hello world";
    // cout<<st;

    // taking inputs 
        // int integer;
        // cin>>integer;
        // double double_precision;
        // cin>>double_precision;
        // char ch;
        // cin>>ch;
        // bool b;
        // cin>>b;
        // cout<<integer<<double_precision<<ch<<b;

    // string str;
    // cin>>str;//only reads one token
    // getline(cin,str);
    // cout<<str;

    // typecasting
    // int n=(int)3.3;
    // cout<<n;
    // double d=(double) 4;
    // cout<<d;

    // int n=333;
    // string s=to_string(n);
    // cout<<s;
    // string s="3";
    // int n=stoi(s);
    // cout<<n;
    // string d="3.33";
    // double m=stod(d);
    // cout<<m;

    // cout<<factorial(5);
    // cout<<factorial(-3);
    // cout<<factorial(0);
    // vector<int> vec;
    // vec.reserve(100);
    
    // // cout<<vec.size(); 0
    // vec.push_back(3);
    // vec.emplace_back(34);
    // // vec.pop_back();
    // vec.emplace(vec.begin()+1,33);
    // vec.emplace(vec.begin(),9);
    // vec.insert(vec.begin(),99);
    // vec.erase(vec.begin());
    // cout<<vec[3];

    // for(auto value: vec)cout<<value<<endl;

    // map<int,int> hashmap;
    // hashmap[3]=4;
    // hashmap[4]=8;
    // hashmap[5]=5;
    // cout<<hashmap[99];//returns 0 if key not there
    
    return 0;
}
long factorial(int positive_number){
    if (positive_number<0) return -1;
    long result=1;
    for(int i=1;i<=positive_number;i++)result*=i;
    return result;
}